docs/deployment/azure/application-insights.md

```diff
@@ -9,60 +9,123 @@ ms.topic: how-to
 
-var cache = builder.AddRedis("cache");
+var insights = builder.AddConnectionString("myInsightsResource", "APPLICATIONINSIGHTS_CONNECTION_STRING");
 
-var apiservice = builder.AddProject<Projects.AspireApp_ApiService>("apiservice")
-    .WithEnvironment(
-        "APPLICATIONINSIGHTS_CONNECTION_STRING",
-        appInsightsConnectionString);
+var apiService = builder.AddProject<Projects.ApiService>("apiservice")
+    .WithReference(insights);
 
-builder.AddProject<Projects.AspireApp_Web>("webfrontend")
-    .WithReference(cache)
-    .WithReference(apiservice)
-    .WithEnvironment(
-        "APPLICATIONINSIGHTS_CONNECTION_STRING",
-        appInsightsConnectionString);
+builder.AddProject<Projects.Web>("webfrontend")
+    .WithReference(apiService)
+    .WithReference(insights);
 
 builder.Build().Run();
 ```
 
-The preceding reads the connection string from configuration, as this is a secret we should store the value in [app secrets](/aspnet/core/security/app-secrets). Right click on the AppHost project and choose Manage Secrets from the context menu to open the secrets file for the AppHost project. In the file add the key and your specific connection string, the example below is for illustration purposes.
+#### Resource usage during developemnt
+
+When running the .NET Aspire app locally, the preceding code reads the connection string from configuration. As this is a secret, you should store the value in [app secrets](/aspnet/core/security/app-secrets). Right click on the app host project and choose **Manage Secrets** from the context menu to open the secrets file for the app host project. In the file add the key and your specific connection string, the example below is for illustration purposes.
 
 ```json
 {
-  "APPLICATIONINSIGHTS_CONNECTION_STRING": "InstrumentationKey=12345678-abcd-1234-abcd-1234abcd5678;IngestionEndpoint=https://westus3-1.in.applicationinsights.azure.com"
+  "ConnectionStrings": {
+    "myInsightsResource": "InstrumentationKey=12345678-abcd-1234-abcd-1234abcd5678;IngestionEndpoint=https://westus3-1.in.applicationinsights.azure.com"
+  }
 }
 ```
 
+> [!NOTE]
+> The `name` specified in the app host code needs to match a key inside the `ConnectionStrings` section in the settings file.
+
+#### Resource usage during deployment
+
+When [deploying an Aspire application with Azure Developer CLI (`azd`)](./aca-deployment-azd-in-depth.md), it will recognize the connection string resource and prompt for a value. This enables a different resource to be used for the deployment from the value used for local development.
+
+### Mixed deployment
+
+If you wish to use a different deployment mechanism per execution context, use the appropriate API conditionally. For example, the following code uses a pre-supplied connection at development time, and an automatically provisioned resource at deployment time.
+
+``` csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var insights = builder.ExecutionContext.IsPublishMode
```

Comment:
Would it be better to have the condition around the other way. I would imagine that in production deployments you want to use a specific app insights instance rather than one that is randomly created (if you care about those things).

Reply:
That's a good question. I can see both sides...

Reply:
It was done this way because we are not talking about design-time provisioning in p4. Once that is in we can flip it back.